src/main/scala/mesosphere/marathon/upgrade/ScalingProposition.scala

```diff
@@ -0,0 +1,9 @@
+package mesosphere.marathon.upgrade
+
+import mesosphere.marathon.Protos.MarathonTask
+
+class ScalingProposition(private val candidatesToKill: Seq[MarathonTask],
+                         private val numberOfTasksToStart: Int) {
+  val tasksToKill = if (candidatesToKill.nonEmpty) Some(candidatesToKill) else None
```

Comment:
I think that I would use a case class and directly calculate the `Option`s in `scalingProposition` method.